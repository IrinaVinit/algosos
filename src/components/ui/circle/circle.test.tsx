import renderer from "react-test-renderer";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

const letter ="T";
const string = 'string'
const element = <div><h1>element</h1></div>
const number = 5;


describe("test Circle component", ()=>{
    it("render without letter", ()=>{
        const tree = renderer.create(<Circle />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("render with letter", ()=>{
        const tree = renderer.create(<Circle letter={letter}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("render with head-string", ()=>{
        const tree = renderer.create(<Circle head={string}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("render with head-element", ()=>{
        const tree = renderer.create(<Circle head={element}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("render with tail-string", ()=>{
        const tree = renderer.create(<Circle tail={string}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("render with tail-element", ()=>{
        const tree = renderer.create(<Circle tail={element}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("render with index", ()=>{
        const tree = renderer.create(<Circle index={number}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("render with isSmall", ()=>{
        const tree = renderer.create(<Circle isSmall/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("render with default state", ()=>{
        const tree = renderer.create(<Circle state={ElementStates.Default}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("render with changing state", ()=>{
        const tree = renderer.create(<Circle state={ElementStates.Changing}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("render with modified state", ()=>{
        const tree = renderer.create(<Circle state={ElementStates.Modified}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
