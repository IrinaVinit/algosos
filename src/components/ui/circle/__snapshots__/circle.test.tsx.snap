// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Circle component render with changing state 1`] = `
<div
  className="content "
  data-cy="circle-content"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render with default state 1`] = `
<div
  className="content "
  data-cy="circle-content"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render with head-element 1`] = `
<div
  className="content "
  data-cy="circle-content"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div>
      <h1>
        element
      </h1>
    </div>
  </div>
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render with head-string 1`] = `
<div
  className="content "
  data-cy="circle-content"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    string
  </div>
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render with index 1`] = `
<div
  className="content "
  data-cy="circle-content"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    5
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`test Circle component render with isSmall 1`] = `
<div
  className="content "
  data-cy="circle-content"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render with letter 1`] = `
<div
  className="content "
  data-cy="circle-content"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      T
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render with modified state 1`] = `
<div
  className="content "
  data-cy="circle-content"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render with tail-element 1`] = `
<div
  className="content "
  data-cy="circle-content"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div>
      <h1>
        element
      </h1>
    </div>
  </div>
</div>
`;

exports[`test Circle component render with tail-string 1`] = `
<div
  className="content "
  data-cy="circle-content"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    string
  </div>
</div>
`;

exports[`test Circle component render without letter 1`] = `
<div
  className="content "
  data-cy="circle-content"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
